### DISCUSSION TITLE
Multiple search fields and models

### DISCUSSION THREAD

**Discussion by pepukrit** [carltongibson/django-filter#1428] [Q&A]
Hi, all. I'm looking for a way to filter two inputs(name and place) from two models (Photographer and Location); but, MultipleChoiceFilter seems doesn't work for me because I need to add more complex logic which performs in my custom filter.

    class SearchResultFilter(filters.FilterSet):
      name = filters.CharFilter('Photographer')
      place = filters.CharFilter(method='location_filter', distinct=True)

      class Meta:
         model = models.Photographer
         fields = ('name')

      def location_filter(self, queryset, name, value):
         """My complex logic"""
         .
         .
         .
         model = models.Location.filter(Q(location__icontains=value))
         return model

The question is " Is there a way I can put **name** input field in my location_filter, so that I can filter two inputs (name and place) in this function and return a wanted querySet from here".

I tried to use self.name in location_filter; however, an error log produced. I guess **"name"** here is not an class variable

**Comment by kjpc-tech** [comment]
I'm in a similar scenario of trying to filter by two dependant fields (`postal_code` and `radius`) inside of one method. 

```
class MyFilter(filters.FilterSet):
  postal_code = filters.CharFilter()
  radius = filters.NumberFilter(method='location_filter')

  class Meta:
     model = models.MyModel
     fields = ('name', 'etc')

  def location_filter(self, queryset, name, value):
     lat, lng = get_lat_lng_from_postal(postal)
     radius = value
     return queryset.filter(
         location__distance_lt=(Point(lng, lat), Distance(mi=radius))
     )
```

Is there some simple way to have one filter method use multiple filter fields at the same time? Is there some recommended way to handle a scenario like this?

**Reply by kjpc-tech** [reply]
I ended up doing this: https://gitlab.com/-/snippets/2237049
